Compilation error in logger_message.h with Linux, Clang++

When trying to compile kratos, with the following options for CMake,
```
-DCMAKE_BUILD_TYPE=Debug
-DCMAKE_C_COMPILER=/usr/bin/clang-6.0
-DCMAKE_CXX_COMPILER=/usr/bin/clang++-6.0
-DCMAKE_CXX_FLAGS= -msse3 -std=c++11
```

I get the following error:
```
In file included from /home/mraschi/Projects/Kratos/applications/StructuralMechanicsApplication/custom_processes/spr_error_process.cpp:18:
In file included from /home/mraschi/Projects/Kratos/kratos/processes/find_nodal_neighbours_process.h:29:
In file included from /home/mraschi/Projects/Kratos/kratos/processes/process.h:29:
In file included from /home/mraschi/Projects/Kratos/kratos/includes/kratos_flags.h:26:
In file included from /home/mraschi/Projects/Kratos/kratos/containers/flags.h:30:
In file included from /home/mraschi/Projects/Kratos/kratos/includes/serializer.h:34:
In file included from /home/mraschi/Projects/Kratos/kratos/input_output/logger.h:23:
/home/mraschi/Projects/Kratos/kratos/input_output/logger_message.h:195:12: error: call to function 'operator<<' that is neither visible in the template definition nor found by argument-dependent lookup
                                buffer << rValue;
                                       ^
/home/mraschi/Projects/Kratos/kratos/input_output/logger.h:151:21: note: in instantiation of function template specialization 'Kratos::LoggerMessage::operator<<<std::vector<double, std::allocator<double> > >' requested here
                  mCurrentMessage << rValue;
                                  ^
/home/mraschi/Projects/Kratos/applications/StructuralMechanicsApplication/custom_processes/spr_error_process.cpp:158:74: note: in instantiation of function template specialization 'Kratos::Logger::operator<<<std::vector<double, std::allocator<double> > >' requested here
        KRATOS_INFO_IF("SPRErrorProcess", mEchoLevel > 2) << "Error GP:" << error_integration_point << std::endl;
                                                                         ^
/home/mraschi/Projects/Kratos/kratos/utilities/stl_io.h:27:15: note: 'operator<<' should be declared prior to the call site
std::ostream& operator<<(std::ostream& rOStream, const std::vector<T>& rVec) {
```

--- a/applications/DelaunayMeshingApplication/delaunay_meshing_application_variables.cpp
+++ b/applications/DelaunayMeshingApplication/delaunay_meshing_application_variables.cpp
@@ -8,7 +8,7 @@
 //
 
 #include "delaunay_meshing_application_variables.h"
-#include "utilities/stl_io.h"
+#include "utilities/stl_vector_io.h"
 #include "containers/global_pointers_vector.h"
 
 namespace Kratos

--- a/kratos/containers/variable.h
+++ b/kratos/containers/variable.h
@@ -2,15 +2,15 @@
 //    ' /   __| _` | __|  _ \   __|
 //    . \  |   (   | |   (   |\__ `
 //   _|\_\_|  \__,_|\__|\___/ ____/
-//                   Multi-Physics 
+//                   Multi-Physics
 //
-//  License:		 BSD License 
+//  License:		 BSD License
 //					 Kratos default license: kratos/license.txt
 //
 //  Main authors:    Pooyan Dadvand
 //                   Riccardo Rossi
 //  Collaborator:    Vicente Mataix Ferrandiz
-//                    
+//
 //
 
 #if !defined(KRATOS_VARIABLE_H_INCLUDED )
@@ -25,7 +25,7 @@
 // Project includes
 #include "includes/define.h"
 #include "variable_data.h"
-#include "utilities/stl_io.h"
+#include "utilities/stl_vector_io.h"
 
 namespace Kratos
 {
@@ -80,8 +80,8 @@ public:
     ///@name Life Cycle
     ///@{
 
-    /** 
-     * Constructor with specific name and zero value 
+    /**
+     * Constructor with specific name and zero value
      * @param NewName The name to be assigned to the new variable
      * @param Zero The value to be assigned to the variable as zero. In case of not definition will take the value given by the constructor of the time
      */
@@ -102,7 +102,7 @@ public:
     ///@}
     ///@name Operators
     ///@{
-    
+
     /**
      * Assignment operator.
      * @param rOtherVariable The old variable to be assigned
@@ -119,8 +119,8 @@ public:
     ///@{
 
     /**
-     * Clone creates a copy of the object using a copy constructor of the class. 
-     * It is useful to avoid shallow copying of complex objects and also without 
+     * Clone creates a copy of the object using a copy constructor of the class.
+     * It is useful to avoid shallow copying of complex objects and also without
      * actually having information about the variable type.
      * @param pSource The pointer of the variable to be cloned
      * @return A raw pointer of the variable
@@ -131,8 +131,8 @@ public:
     }
 
     /**
-     * Copy is very similar to Clone except that it also the destination 
-     * pointer also passed to it. It is a helpful method specially 
+     * Copy is very similar to Clone except that it also the destination
+     * pointer also passed to it. It is a helpful method specially
      * to create a copy of heterogeneous data arrays
      * @param pSource The pointer of the variable to be copied
      * @param pDestination The pointer of the destination variable
@@ -144,9 +144,9 @@ public:
     }
 
     /**
-     * Assign is very similar to Copy. It just differs in using an assignment 
-     * operator besides the copy constructor. Copy creates a new object while 
-     * Assign does the assignment for two existing objects. 
+     * Assign is very similar to Copy. It just differs in using an assignment
+     * operator besides the copy constructor. Copy creates a new object while
+     * Assign does the assignment for two existing objects.
      * @param pSource The pointer of the value to be assigned
      * @param pDestination The pointer of the destination value
      */
@@ -156,7 +156,7 @@ public:
     }
 
     /**
-     * AssignZero is a special case of Assign for which variable zero value used as source. 
+     * AssignZero is a special case of Assign for which variable zero value used as source.
      * This method is useful for initializing arrays or resetting values in memory.
      * @param pDestination The pointer of the destination variable
      */
@@ -167,8 +167,8 @@ public:
     }
 
     /**
-     *  Delete removes an object of variable type from memory. It calls a 
-     * destructor of objects to prevent memory leak and frees the memory 
+     *  Delete removes an object of variable type from memory. It calls a
+     * destructor of objects to prevent memory leak and frees the memory
      * allocated for this object assuming that the object is allocated in heap.
      * @param pSource The pointer of the variable to be deleted
      */
@@ -178,8 +178,8 @@ public:
     }
 
     /**
-     *  Destruct eliminates an object maintaining the memory it is using. 
-     * However, the unlike Delete it does nothing with the memory allocated to it. 
+     *  Destruct eliminates an object maintaining the memory it is using.
+     * However, the unlike Delete it does nothing with the memory allocated to it.
      * So it is very useful in case of reallocating a part of the memory.
      * @param pSource The pointer of the variable to be destructed
      */
@@ -189,9 +189,9 @@ public:
     }
 
     /**
-     *  Print is an auxiliary method to produce output of given variable 
-     * knowing its address. For example writing an heterogenous container 
-     * in an output stream can be done using this method. Point assumes 
+     *  Print is an auxiliary method to produce output of given variable
+     * knowing its address. For example writing an heterogenous container
+     * in an output stream can be done using this method. Point assumes
      * that the streaming operator is defined for the variable type.
      * @param pSource The pointer of the variable to be printed
      * @param rOStream The stream used to print the information
@@ -202,9 +202,9 @@ public:
     }
 
     /**
-     * PrintData is an auxiliary method to produce output only the value of given variable 
-     * knowing its address. For example writing an heterogenous container 
-     * in an output stream can be done using this method. Point assumes 
+     * PrintData is an auxiliary method to produce output only the value of given variable
+     * knowing its address. For example writing an heterogenous container
+     * in an output stream can be done using this method. Point assumes
      * that the streaming operator is defined for the variable type.
      * @param pSource The pointer of the variable to be printed
      * @param rOStream The stream used to print the information
@@ -275,7 +275,7 @@ public:
     ///@name Input and output
     ///@{
 
-    /** 
+    /**
      * Turn back information as a string.
      */
     std::string Info() const override
@@ -320,7 +320,7 @@ protected:
     ///@name Protected Operators
     ///@{
 
- 
+
     ///@}
     ///@name Protected Operations
     ///@{
@@ -446,6 +446,6 @@ inline std::ostream& operator << (std::ostream& rOStream,
 
 }  // namespace Kratos.
 
-#endif // KRATOS_VARIABLE_H_INCLUDED  defined 
+#endif // KRATOS_VARIABLE_H_INCLUDED  defined
 
 

--- a/kratos/includes/exception.h
+++ b/kratos/includes/exception.h
@@ -24,6 +24,7 @@
 // Project includes
 #include "includes/kratos_export_api.h"
 #include "includes/code_location.h"
+#include "utilities/stl_io.h"
 
 
 namespace Kratos

--- a/kratos/includes/exception.h
+++ b/kratos/includes/exception.h
@@ -24,7 +24,7 @@
 // Project includes
 #include "includes/kratos_export_api.h"
 #include "includes/code_location.h"
-#include "utilities/stl_io.h"
+#include "utilities/stl_io.h" // Defines operator<< for several stl types, needed to print such types in error messages using Exception::operator<<
 
 
 namespace Kratos

--- a/kratos/includes/exception.h
+++ b/kratos/includes/exception.h
@@ -24,7 +24,7 @@
 // Project includes
 #include "includes/kratos_export_api.h"
 #include "includes/code_location.h"
-#include "utilities/stl_io.h" // Defines operator<< for several stl types, needed to print such types in error messages using Exception::operator<<
+#include "utilities/stl_vector_io.h"
 
 
 namespace Kratos

--- a/kratos/includes/shared_pointers.h
+++ b/kratos/includes/shared_pointers.h
@@ -15,6 +15,7 @@
 #define  KRATOS_MEMORY_H_INCLUDED
 
 /* System includes */
+#include <iostream>
 #include <utility>
 
 /* External includes */
@@ -45,6 +46,23 @@ template<typename C, typename...Args>
 unique_ptr<C> make_unique(Args &&...args) {
     // Note: std::make_unique is C++14, this can be updated once we upgrade from C++11
     return unique_ptr<C>(new C(std::forward<Args>(args)...));
+}
+
+template<class T>
+std::ostream& operator <<(std::ostream& rOStream, const Kratos::weak_ptr<T>& rData) {
+
+  if(!rData.expired())
+    rOStream << *rData.lock().get();
+  else
+    rOStream <<" expired weak_ptr ";
+
+  return rOStream;
+}
+
+template<class T>
+std::ostream& operator <<(std::ostream& rOStream, const Kratos::intrusive_ptr<T>& rData) {
+  rOStream << *rData.get();
+  return rOStream;
 }
 
 } // namespace Kratos

--- a/kratos/input_output/logger_message.h
+++ b/kratos/input_output/logger_message.h
@@ -31,6 +31,7 @@
 // Project includes
 #include "includes/kratos_export_api.h"
 #include "includes/code_location.h"
+#include "utilities/stl_vector_io.h"
 
 namespace Kratos
 {

--- a/kratos/tests/cpp_tests/sources/test_checks.cpp
+++ b/kratos/tests/cpp_tests/sources/test_checks.cpp
@@ -12,7 +12,7 @@
 //
 
 // System includes
-
+#include <vector>
 
 // External includes
 
@@ -101,6 +101,8 @@ namespace Kratos {
 
         KRATOS_TEST_CASE_IN_SUITE(VectorChecks, KratosCoreFastSuite)
         {
+            using Kratos::operator<<; // operator<< for vectors is defined in namespace Kratos, we are inside the namespace "Testing"
+
             std::vector<double> v1{1.0, 2.0};
             std::vector<double> v2{1.0, 1.99};
             std::vector<double> v3{1.0, 2.0, 3.0};

--- a/kratos/tests/cpp_tests/sources/test_checks.cpp
+++ b/kratos/tests/cpp_tests/sources/test_checks.cpp
@@ -101,8 +101,6 @@ namespace Kratos {
 
         KRATOS_TEST_CASE_IN_SUITE(VectorChecks, KratosCoreFastSuite)
         {
-            using Kratos::operator<<; // operator<< for vectors is defined in namespace Kratos, we are inside the namespace "Testing"
-
             std::vector<double> v1{1.0, 2.0};
             std::vector<double> v2{1.0, 1.99};
             std::vector<double> v3{1.0, 2.0, 3.0};

--- a/kratos/tests/cpp_tests/utilities/test_stl_io.cpp
+++ b/kratos/tests/cpp_tests/utilities/test_stl_io.cpp
@@ -14,8 +14,8 @@
 
 // Project includes
 #include "testing/testing.h"
-#include "utilities/stl_io.h"
-
+#include "utilities/stl_vector_io.h"
+#include "includes/shared_pointers.h"
 
 namespace Kratos {
 namespace Testing {
@@ -39,27 +39,6 @@ KRATOS_TEST_CASE_IN_SUITE(StdVectorOutputStream, KratosCoreFastSuite)
     std::stringstream ss_string;
     ss_string << string_vector;
     KRATOS_CHECK_STRING_EQUAL(ss_string.str(), "[val_1, custom, again_test]");
-}
-
-KRATOS_TEST_CASE_IN_SUITE(StdSetOutputStream, KratosCoreFastSuite)
-{
-    using Kratos::operator<<; // needed bcs it is inside the namespace "Testing"
-
-    std::set<int> int_set {1, 5,-63, 581, 6};
-    std::set<double> double_set {4.335, 8.639, -888.47, 9874.0};
-    std::set<std::string> string_set {"short", "middle", "largest"};
-
-    std::stringstream ss_int;
-    ss_int << int_set;
-    KRATOS_CHECK_STRING_EQUAL(ss_int.str(), "[-63, 1, 5, 6, 581]");
-
-    std::stringstream ss_double;
-    ss_double << double_set;
-    KRATOS_CHECK_STRING_EQUAL(ss_double.str(), "[-888.47, 4.335, 8.639, 9874]");
-
-    std::stringstream ss_string;
-    ss_string << string_set;
-    KRATOS_CHECK_STRING_EQUAL(ss_string.str(), "[largest, middle, short]");
 }
 
 KRATOS_TEST_CASE_IN_SUITE(StdWeakPtrOutputStream, KratosCoreFastSuite)

--- a/kratos/utilities/kratos_log.h
+++ b/kratos/utilities/kratos_log.h
@@ -2,13 +2,13 @@
 //    ' /   __| _` | __|  _ \   __|
 //    . \  |   (   | |   (   |\__ `
 //   _|\_\_|  \__,_|\__|\___/ ____/
-//                   Multi-Physics 
+//                   Multi-Physics
 //
-//  License:		 BSD License 
+//  License:		 BSD License
 //					 Kratos default license: kratos/license.txt
 //
 //  Main authors:    Pooyan Dadvand
-//                    
+//
 //
 
 
@@ -30,11 +30,11 @@
 
 // Project includes
 #include "includes/define.h"
-#include "utilities/stl_io.h"
+#include "utilities/stl_vector_io.h"
 
 // Filters
 #define KRATOS_LOG_ADD_CUSTOM_FILTER(STRING) KratosLogUtils::GetInstance().AddCustomFilter(STRING);
-#define KRATOS_LOG_FILTER(STRING)            KratosLogUtils::GetInstance().Filter(STRING)  
+#define KRATOS_LOG_FILTER(STRING)            KratosLogUtils::GetInstance().Filter(STRING)
 
 // Stamps
 #define KRATOS_TIME_STAMP "[" << KratosLogUtils::GetInstance().CurrentDateTime() << "]"
@@ -100,17 +100,17 @@
 #define KRATOS_LOG_TRACE_IF_N(C,N)    KRATOS_LOG_IF_N(KRATOS_SEVERITY_TRACE,C)      << KRATOS_LOG_TRACE_STAMP
 #define KRATOS_LOG_TRACE_FIRST_N(N)   KRATOS_LOG_FIRST_N(KRATOS_SEVERITY_TRACE,N)   << KRATOS_LOG_TRACE_STAMP
 #else
-#define KRATOS_LOG_DEBUG              
-#define KRATOS_LOG_DEBUG_N(N)         
-#define KRATOS_LOG_DEBUG_IF(C)        
-#define KRATOS_LOG_DEBUG_IF_N(C,N)    
-#define KRATOS_LOG_DEBUG_FIRST_N(N)   
+#define KRATOS_LOG_DEBUG
+#define KRATOS_LOG_DEBUG_N(N)
+#define KRATOS_LOG_DEBUG_IF(C)
+#define KRATOS_LOG_DEBUG_IF_N(C,N)
+#define KRATOS_LOG_DEBUG_FIRST_N(N)
 
-#define KRATOS_LOG_TRACE              
-#define KRATOS_LOG_TRACE_N(N)         
-#define KRATOS_LOG_TRACE_IF(C)        
-#define KRATOS_LOG_TRACE_IF_N(C,N)    
-#define KRATOS_LOG_TRACE_FIRST_N(N)   
+#define KRATOS_LOG_TRACE
+#define KRATOS_LOG_TRACE_N(N)
+#define KRATOS_LOG_TRACE_IF(C)
+#define KRATOS_LOG_TRACE_IF_N(C,N)
+#define KRATOS_LOG_TRACE_FIRST_N(N)
 #endif
 
 // Normal log
@@ -125,19 +125,19 @@
   if (++KRATOS_LOG_OCCURRENCES > __N__) KRATOS_LOG_OCCURRENCES -= __N__; \
   if (KRATOS_LOG_OCCURRENCES == 1) \
     KRATOS_LOG(KRATOS_SEVERITY)
-    
+
 // If block
 #define KRATOS_LOG_IF(KRATOS_SEVERITY,__C__) \
   if (__C__) \
     KRATOS_LOG(KRATOS_SEVERITY)
-    
+
 // If-Each-N block
 #define KRATOS_LOG_IF_N(KRATOS_SEVERITY,__C__,__N__) \
   static int KRATOS_LOG_OCCURRENCES = 0;\
   if (++KRATOS_LOG_OCCURRENCES > __N__) KRATOS_LOG_OCCURRENCES -= __N__; \
   if ((__C__) && KRATOS_LOG_OCCURRENCES == 1) \
     KRATOS_LOG(KRATOS_SEVERITY)
-    
+
 // First-N block
 #define KRATOS_LOG_FIRST_N(KRATOS_SEVERITY,__N__) \
   static int KRATOS_LOG_OCCURRENCES = 0;\
@@ -165,7 +165,7 @@ namespace Kratos
 ///@}
 ///@name Kratos Classes
 ///@{
-  
+
 enum KRATOS_SEVERITY {
   KRATOS_SEVERITY_ERROR,
   KRATOS_SEVERITY_WARNING,
@@ -175,60 +175,60 @@ enum KRATOS_SEVERITY {
   KRATOS_SEVERITY_TRACE,
   KRATOS_LOG_SOLVER
 };
- 
-template<KRATOS_SEVERITY S>  
+
+template<KRATOS_SEVERITY S>
 class KratosLog : public std::ostream
 {
   private:
-    
+
     KratosLog<S>();
-    
+
     static bool         mInstanceFlag;
     static KratosLog *  mpInstance;
-    
+
   public:
-         
+
     ~KratosLog<S>();
-    
+
     static KratosLog<S>& GetInstance();
-    
+
     void SetSeverityLevel(int severity);
     void SetOutput(std::string const& LogFileName);
-    
+
     int getSeverity();
     std::ostream& getOutput();
-    
+
     bool            mPrintLogOnScreen;
     int             mSeverityLevel;
     std::ofstream * mOutputFile;
-    
-};  
+
+};
 
 class KratosLogUtils
 {
   private:
-  
+
     KratosLogUtils();
-    
+
     static bool mInstanceFlag;
     static KratosLogUtils * mpInstance;
-    
+
     std::vector<std::string> * mFilterString;
-    
+
   public:
-    
+
     ~KratosLogUtils();
-    
+
     static KratosLogUtils& GetInstance();
-    
+
     void AddCustomFilter(const std::string filter);
-    
+
     const std::string CurrentDateTime();
     const std::string CurrentDateTime(const char * format);
     const char * Filter(const char * inputString);
     const char * FilterNamespace(const char * inputString);
     const char * FilterCustom(const char * inputString, std::vector<std::string> remove);
-    
+
 };
 
 ///@}
@@ -240,7 +240,7 @@ class KratosLogUtils
 ///@}
 ///@name Input and output
 ///@{
-  
+
 // Generic Specialization
 template<KRATOS_SEVERITY S>
 KratosLog<S>::KratosLog()
@@ -265,33 +265,33 @@ void KratosLog<S>::SetOutput(std::string const& LogFileName)
 }
 
 template<KRATOS_SEVERITY S>
-KratosLog<S>& KratosLog<S>::GetInstance() 
+KratosLog<S>& KratosLog<S>::GetInstance()
 {
     if(!mInstanceFlag)
-    {   
+    {
         mInstanceFlag = 1;
         mpInstance = new KratosLog<S>();
         mpInstance->SetOutput("KratosLog.log");
     }
-    
+
     return (*mpInstance);
 }
 
 // Error specialization
 inline KratosLog<KRATOS_SEVERITY_ERROR> & operator << (KratosLog<KRATOS_SEVERITY_ERROR> &out, std::ostream& (*fn)(std::ostream&))
-{ 
+{
   if(out.mPrintLogOnScreen)
     fn(std::cout);
-  
+
   fn((*out.mOutputFile));
   (*out.mOutputFile).flush();
-  
+
     abort();
   //TODO: Throw exception abort();
 
   return out;
 }
-      
+
 template<class T>
 inline KratosLog<KRATOS_SEVERITY_ERROR> & operator << (KratosLog<KRATOS_SEVERITY_ERROR> &out, const T& data)
 {
@@ -301,25 +301,25 @@ inline KratosLog<KRATOS_SEVERITY_ERROR> & operator << (KratosLog<KRATOS_SEVERITY
   }
 
   (*out.mOutputFile) << data;
-  
+
   //TODO: Throw exception abort();
-  
-  return out;  
+
+  return out;
 }
-    
+
 // Error specialization
 template<KRATOS_SEVERITY S>
 inline KratosLog<S> & operator << (KratosLog<S> &out, std::ostream& (*fn)(std::ostream&))
-{ 
+{
   if(out.mPrintLogOnScreen)
     fn(std::cout);
-  
+
   fn((*out.mOutputFile));
   (*out.mOutputFile).flush();
 
   return out;
 }
-  
+
 template<class T, KRATOS_SEVERITY S>
 inline KratosLog<S> & operator << (KratosLog<S> &out, const T& data)
 {
@@ -329,8 +329,8 @@ inline KratosLog<S> & operator << (KratosLog<S> &out, const T& data)
   }
 
   (*out.mOutputFile) << data;
-  
-  return out;  
+
+  return out;
 }
 
 ///@}
@@ -338,6 +338,6 @@ inline KratosLog<S> & operator << (KratosLog<S> &out, const T& data)
 
 }  // namespace Kratos.
 
-#endif // KRATOS_KRATOS_LOG_H_INCLUDED  defined 
+#endif // KRATOS_KRATOS_LOG_H_INCLUDED  defined
 
 

--- a/kratos/utilities/stl_io.h
+++ b/kratos/utilities/stl_io.h
@@ -18,7 +18,6 @@
 #include <iostream>
 #include <vector>
 #include <set>
-#include <memory>
 
 // Project includes
 #include "includes/define.h"
@@ -54,23 +53,6 @@ std::ostream& operator <<(std::ostream& rOStream, const std::set<T>& rSet) {
   }
   rOStream << "]";
 
-  return rOStream;
-}
-
-template<class T>
-std::ostream& operator <<(std::ostream& rOStream, const std::weak_ptr<T>& rData) {
-
-  if(!rData.expired())
-    rOStream << *rData.lock().get();
-  else
-    rOStream <<" expired weak_ptr ";
-
-  return rOStream;
-}
-
-template<class T>
-std::ostream& operator <<(std::ostream& rOStream, const Kratos::intrusive_ptr<T>& rData) {
-  rOStream << *rData.get();
   return rOStream;
 }
 

--- a/kratos/utilities/stl_vector_io.h
+++ b/kratos/utilities/stl_vector_io.h
@@ -11,13 +11,12 @@
 //
 //
 
-#if !defined(KRATOS_KRATOS_STL_IO_H_INCLUDED )
-#define  KRATOS_KRATOS_STL_IO_H_INCLUDED
+#if !defined(KRATOS_STL_VECTOR_IO_H_INCLUDED )
+#define  KRATOS_STL_VECTOR_IO_H_INCLUDED
 
 // System includes
 #include <iostream>
 #include <vector>
-#include <set>
 
 // Project includes
 #include "includes/define.h"
@@ -40,23 +39,7 @@ std::ostream& operator<<(std::ostream& rOStream, const std::vector<T>& rVec) {
     return rOStream;
 }
 
-template<class T>
-std::ostream& operator <<(std::ostream& rOStream, const std::set<T>& rSet) {
-
-  const std::size_t set_size = rSet.size();
-
-  rOStream << "[";
-  if(set_size>0) rOStream << *(rSet.begin());
-  if(set_size>1) {
-    for(auto it(std::next(rSet.begin(),1)); it!=rSet.end(); ++it)
-      rOStream<<", "<<*it;
-  }
-  rOStream << "]";
-
-  return rOStream;
-}
-
 } //namespace Kratos
 
 
-#endif // KRATOS_KRATOS_STL_IO_H_INCLUDED  defined
+#endif // KRATOS_STL_VECTOR_IO_H_INCLUDED  defined

